src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/table_view.rs

```diff
@@ -0,0 +1,198 @@
+                      result += ("<th>" + elt + "</th>");
+                    });
+                } else {
+                    data[0].forEach(function (elt, ix) {
+                      result += ("<th>" + ix + "</th>");
+                    });
+                };
+                result += "</tr>";
+                table = []
+
+                data.forEach(function(d, i) {
+                  d.forEach(function(elem, idx) {
+                    table[idx] = table[idx] || []
+                    table[idx].push(elem)
+                  })
+                })
+
+                table.forEach(function (row, ix) {
+                  result += ("<tr><th>" + ix + "</th>");
+                  row.forEach(function (d) {
+                    result += toTableCell(d, level);
+                  });
+                  result += ("</tr>")
+                });
+                return tableOf(result, level);
+            }
+
+            var genGenericTable = function (data, level) {
+                var result = "";
+                data.forEach(function (point, ix) {
+                  result += ("<tr><th>" + ix + "</th>" + toTableCell(point, level) + "</tr>");
+                });
+                return tableOf(result, level);
+            }
+
+            var genRowObjectTable = function (data, level) {
+                var keys = Object.keys(data);
+                var result = "<tr>";
+                keys.forEach(function (key) {
+                  result += ("<th>" + key + "</th>");
+                });
+                result += "</tr><tr>";
+                keys.forEach(function (key) {
+                  result += toTableCell(data[key], level);
+                });
+                result += "</tr>";
+                return tableOf(result, level);
+            }
+
+            var toTableCell = function (data, level) {
+                if (Array.isArray(data)) {
+                  return "<td>" + genTable(data, level + 1) + "</td>";
+                } else if (data instanceof Object) {
+                  return "<td>" + genRowObjectTable(data, level + 1) + "</td>";
+                } else {
+                  if (data === undefined || data === null) data = "";
+                  var res = data.toString();
+                  return '<td class="plaintext">' + (res === "" ? "N/A" : res) + '</td>';
+                }
+            }
+
+            var genTable = function (data, level, header) {
+                if (isMatrix(data)) {
+                  return genMatrix(data, level, header);
+                } else if (isObjectMatrix(data)) {
+                  return genObjectMatrix(data, level);
+                } else {
+                  return genGenericTable(data, level);
+                }
+            }
+
+            while (this.dom.firstChild) {
+                this.dom.removeChild(this.dom.lastChild);
+            }
+    "#;
+
+    let source_style = format!("const style = `{}`;", style());
```

Comment:
The mechanism for setting styles should be also be in js. We want to just move some day a js file to some directory in Stdlib, so all the logic should not be in rust.

Unfortunately that means, that you need to embed the stylesheet into js file.

Reply:
👎 Oh god, actually at first it used to be a js file with a css embedded in it, then i moved it to rust like the bubble chart vis. I'll do it then :)